sqlx-core/src/postgres/options/mod.rs

```diff
@@ -345,3 +356,244 @@ fn default_host(port: u16) -> String {
     // fallback to localhost if no socket was found
     "localhost".to_owned()
 }
+
+/// try to load a password from the various pgpass file locations
+fn load_password(host: &str, port: u16, username: &str, database: Option<&str>) -> Option<String> {
+    #[cfg(not(target_os = "windows"))]
+    let default_file = dirs::home_dir().map(|path| path.join(".pgpass"));
+    #[cfg(target_os = "windows")]
+    let default_file = dirs::data_dir().map(|path| path.join("postgres").join("pgpass.conf"));
+
+    if let Some(file) = default_file {
+        if let Some(password) = load_password_from_file(file, host, port, username, database) {
+            return Some(password);
+        }
+    }
+
+    let custom_file = var_os("PGPASSFILE")?;
```

Comment:
Shouldn't this take precedent over the default path if it is set?

Reply:
yes